transform-plugins/src/main/java/io/cdap/plugin/Hasher.java

```diff
@@ -44,28 +49,27 @@
   private final Config config;
   private Set<String> fieldSet = new HashSet<>();
 
-
   // For testing purpose only.
   public Hasher(Config config) {
     this.config = config;
   }
 
   @Override
-  public void initialize(TransformContext context) throws Exception {
-    super.initialize(context);
-    config.validate();
-    // Split the fields to be hashed.
-    String[] fields = config.fields.split(",");
-    for (String field : fields) {
-      fieldSet.add(field);
-    }
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    config.validate(stageConfigurer.getInputSchema(), stageConfigurer.getFailureCollector());
+    stageConfigurer.getFailureCollector().getOrThrowException();
+    stageConfigurer.setOutputSchema(stageConfigurer.getInputSchema());
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    config.validate(context.getInputSchema(), context.getFailureCollector());
```

Comment:
should throw validation exception using `collector.getOrThrowException()` in prepareRun.